### DISCUSSION TITLE
Inactive Devices

### DISCUSSION THREAD

**Discussion by jacesaces** [aarons22/homebridge-bond#126] [Q&A] [answered]
First of all -- thank you so much for making this plugin -- it's been super useful!

QQ for you... I deleted a device out of the bond app... but I notice it still shows up in homekit via this plugin... Is there a way to flush out or delete old devices that were previously paired to the bond app?

**Comment by aarons22** [comment]
There are a couple ways to do this:
1. Shutdown homebridge, [clear/delete your cachedAccessories file](https://github.com/aarons22/homebridge-bond/wiki/Clearing-Cached-Accessories), start homebridge
   - This is the easiest/quickest approach, but if you have other homebridge plugins configured, you may lose automations, etc.
2. If you are using Homebridge UI (which I highly recommend), you can go into Homebridge settings and look for "Remove Single Cached Accessory". 
   - This is essentially the same as the first option, but it removes the specific device from the cachedAccessories file. You could also do this manually, but I wouldn't recommend it.
3. If either of those options feels like too much to you, you can add the device id to the `hide_device_ids` attribute of your [bond config](https://github.com/aarons22/homebridge-bond#bond-parameters)

**Comment by jacesaces** [comment]
Hi Aaron -- thanks for sending all of these over.

I've been continuing to try to work through these and dig through the
options, but unfortunately all have come up short.

1. I was nervous to try this one due to not wanting to lose automations
(but I did look and was unable to find the cached accessories file)
2. I am using the HOOBS UI. Interestingly, the old bond device isn't
showing up in my list of devices on the HOOBS accessories page anymore (but
it still shows up in Homekit).  I haven't been able to find the setting
called 'remove single cached accessory'
3. This seems like the easiest option... but where can I find the device ID
attribute for the old/removed bond device?

Thanks again for all of your help here!

**Reply by aarons22** [reply]
Hm, if you deleted it from the bond app you won't be able to find the Device Id, so that option probably won't work.

Re-reading your post, I'm going to see if I can replicate the issue on my end by deleting a device. If I can replicate the issue, I may be able to add logic in the plugin to cleanup deleted devices.

I'll report back 😄

**Comment by jacesaces** [comment]
Hmmm that 'inactive' device appears to be crashing homebridge for me now:

4/7/2021, 8:41:56 PM TypeError: Cannot read property 'bondid' of undefined
at /home/hoobs/.hoobs/node_modules/homebridge-bond/dist/platform.js:207:57
at Array.find (<anonymous>) at BondPlatform.bondForDevice
(/home/hoobs/.hoobs/node_modules/homebridge-bond/dist/platform.js:207:37)
at BondPlatform.create
(/home/hoobs/.hoobs/node_modules/homebridge-bond/dist/platform.js:163:27)
at Timeout._onTimeout
(/home/hoobs/.hoobs/node_modules/homebridge-bond/dist/platform.js:155:26)
at listOnTimeout (internal/timers.js:549:17) at processTimers
(internal/timers.js:492:7)

**Comment by jacesaces** [comment, marked as answer]
Ah it was just crashing due to an IP Address change (just changed
routers).  Seems to be working again now.  Would still be great to find out
if we could somehow clear the cache and remove the inactive device, but no
biggie.  Thanks!
